osu.Game/Graphics/UserInterface/SliderAdjustmentTick.cs

```diff
@@ -10,30 +9,22 @@
 
 namespace osu.Game.Graphics.UserInterface
 {
-    public partial class SliderSounds<T> : Component
-        where T : struct, IEquatable<T>, IComparable<T>, IConvertible
+    public partial class SliderAdjustmentTick : Component
     {
         private Sample sample = null!;
-        private double lastSampleTime;
-        private T lastSampleValue;
 
         [BackgroundDependencyLoader(true)]
         private void load(AudioManager audio)
         {
             sample = audio.Samples.Get(@"UI/notch-tick");
         }
 
-        public void PlaySample(T value, float normalizedValue)
+        /// <summary>
+        /// the value by which to adjust the slider tick noise based on the location of the nub on the slider, normalized to be from 0 to 1.
+        /// </summary>
+        /// <param name="normalizedValue"></param>
+        public void PlaySample(float normalizedValue)
         {
-            if (Clock == null || Clock.CurrentTime - lastSampleTime <= 30)
-                return;
```

Comment:
The debounce should have been left in this class. Otherwise the whole point of this class is basically lost.

Reply:
oops. my bad . thanks

Reply:
i don't think its going to work upon testing, the debounce needs to be run before the generic values update is run  on L 192, otherwise sounds often skip themselves.

https://user-images.githubusercontent.com/74463310/213161710-859e4c22-7955-4c7e-b6b1-5a56b43930cb.mp4


https://user-images.githubusercontent.com/74463310/213161740-25e3fe41-6078-498f-9116-725369a7c543.mp4


how does making the method in slidersounds generic and doing everything inside there? , i didnt realise i could do that and it looks plenty clean imo, since its not the whole class being affected

Reply:
simple approach is a no go that requires doing messy things with  the method ( in other words, making it play audio and return a value ) so ill let you handle this one if that's ok